src/snosearch/queries.py

```diff
@@ -1181,6 +1181,35 @@ def __init__(self, params_parser, *args, **kwargs):
         super().__init__(params_parser, *args, **kwargs)
 
 
+class MultipleTypesReportQueryFactory(BasicSearchQueryFactory):
+    '''
+    Like BasicSearchQueryFactory but makes sure item types
+    are specified.
+    '''
+
+    def __init__(self, params_parser, *args, **kwargs):
+        super().__init__(params_parser, *args, **kwargs)
+
+    @assert_something_returned(error_message='Report view requires specifying types:')
+    def _get_item_types(self):
+        return super()._get_item_types()
+
+    def build_query(self):
+        self.validate_item_types()
+        self._get_item_types()
+        return super().build_query()
+
+
+class MultipleTypesReportQueryFactoryWithFacets(MultipleTypesReportQueryFactory, BasicSearchQueryFactoryWithFacets):
+    '''
+    Like BasicSearchQueryFactoryWithFacets but makes sure item types
+    are specified.
+    '''
+
+    def __init__(self, params_parser, *args, **kwargs):
+        super().__init__(params_parser, *args, **kwargs)
+
+
```

Comment:
Can this be simplified to just:
```
class MultipleTypesReportQueryFactoryWithFacets(BasicSearchQueryFactoryWithFacets):

    def __init__(self, params_parser, *args, **kwargs):
        super().__init__(params_parser, *args, **kwargs)
```
Basically it's just an alias over search query with facets. We can drop the `MultipleTypesReportQueryFactory`. I want to be able to do something like `/multitype-report/?query=abc` so we don't need the `Report view requires specifying types` check.